File: LeetCode 热题 100/longestConsecutive.java
import java.util.HashSet;
import java.util.Set;

/*哈希表
 * O(n), O(n)
 *  找出定定数数组中最长连续序列的长度
 * 
 */
class Solution {
    public static int longestConsecutive(int[] nums) {
        Set<Integer> num_set = new HashSet<Integer>();
        for(int num : nums) {
            num_set.add(num);
        }
        int longestStreak = 0;
        for(int num : num_set) {
            if (!num_set.contains(num - 1)) {
                int currenNum = num;
                int currentStreak = 1;
                while (num_set.contains(currenNum + 1)) {
                    currenNum += 1;
                    currentStreak += 1;
                }
                longestStreak = Math.max(longestStreak, currentStreak);
            }
        }
        return longestStreak;
    }



    public static void main(String[] args) {
        int[] nums = {100, 4, 200, 1, 3, 2};
        int longestStreak = longestConsecutive(nums);
        System.out.println("The longest ...");
    }
}

File: LeetCode 热题 100/longestConsecutive.kt
// 根据输入的整数数组，找出最长连续序列的长度
fun longestConsecutive(nums: IntArray): Int {
    // 使用HashSet存储整数数组中的数字，以便快速查找
    val numSet = HashSet(nums)
    var longestStreak = 0
    // 遍历numSet中的每个数字
    for (num in numSet) {
        // 如果当前数字的前一个数字不在numSet中，则存在一个连续序列
        if (!numSet.contains(num - 1)) {
            var currentNum = num
            var currentStreak = 1
            // 继续向后查找连续的数字，直到找到一个不在numSet中的数字
            while (numSet.contains(currentNum + 1)) {
                currentNum += 1
                currentStreak += 1
            }
            // 更新最长序列的长度
            longestStreak = maxOf(longestStreak, currentStreak)
        }
    }
    // 返回最长序列的长度
    return longestStreak
}